### PR TITLE
[wasm] Implement support for pinvoke callbacks in the interpeter.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
Will port to dotnet/runtime manually.

**Comment by vargaz** [comment]
https://github.com/mono/mono/issues/19944

**Comment by vargaz** [comment]
@monojenkins commit apidiff and csproj

**Comment by vargaz** [comment]
@monojenkins commit apidiff and csproj

**Comment by vargaz** [comment]
@monojenkins commit csproj

**Comment by marek-safar** [comment]
@BrzVlad could you review interpreter changes

File: mcs/tools/wasm-tuner/PInvokeTableGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,299 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
```
Comment (SamMonoRT) [inline mcs/tools/wasm-tuner/PInvokeTableGenerator.cs:3]: From PR comments in https://github.com/dotnet/runtime/pull/38664, we want to stick to the 2 line version. We don't need the third line here, please remove

**Comment by vargaz** [comment]
@monojenkins build failed

File: sdks/wasm/src/pinvoke.c [hunk 2]
```diff
@@ -42,3 +42,23 @@ wasm_dl_is_pinvoke_table (void *handle)
 	}
 	return 0;
 }
+
+void foo2(void *p1, void *p2);
```
Comment (BrzVlad) [inline sdks/wasm/src/pinvoke.c:46]: what is this supposed to do ?

Comment (vargaz) [reply sdks/wasm/src/pinvoke.c]: Leftover debug code.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**